src/reader/dataset.rs

```diff
@@ -60,21 +53,16 @@ pub trait ReaderExt: Reader {
             std::mem::size_of::<T>()
         );
 
-        if dsz != std::mem::size_of::<T>() {
-            error!("size of datatype ({}) not same as target {}, alignment may not match and result in unsoundness", dsz, std::mem::size_of::<T>());
-        }
+        ensure!(dsz == std::mem::size_of::<T>(), "size of datatype ({}) not same as target {}, alignment may not match and result in unsoundness", dsz, std::mem::size_of::<T>());
```

Comment:
I am not sure why the previous error was only `log::error` instead of `ensure`, but we have now a failing test which I am not sure how to fix

Reply:
This test reads a string (array of chars). The total size of the string is the number of chars. The best would be to check if std::mem::align_of matches, but we would have to match the HDF5 datatype to the Rust datatype to do that. That is probably why it was only a maybe-error before. Maybe an easier test is to check if dsz*len is a multiple of [alignment](https://doc.rust-lang.org/std/mem/fn.align_of.html)?

Reply:
https://doc.rust-lang.org/reference/type-layout.html#size-and-alignment

It does seem to fix the tests. Not 100% sure if my thinking is correct here.